src/or/control.c

```diff
@@ -3498,7 +3498,7 @@ handle_control_extendcircuit(control_connection_t *conn, uint32_t len,
       connection_printf_to_buf(conn, "552 No such router \"%s\"\r\n", n);
       goto done;
     }
-    if (!node_has_descriptor(node)) {
+    if (!node_has_any_descriptor(node)) {
```

Comment:
The log message at line 3525 is now incorrect:
```
                "controller tried to connect to a node that doesn't have any "
                "addresses that are allowed by the firewall configuration; "
                "circuit marked for closing."
```

We can fail because we are missing the preferred descriptor for the node, too.
Should we do `int first_node = zero_circ;` `first_node = 0;` in this loop, too?

Reply:
e480c11097de79 does it here.